### ISSUE TITLE
[Bug]: Flatpak updates no longer possible with PPA build: Could not unmount revokefs-fuse filesystem

### ISSUE THREAD

**Issue by MikeNavy** [flatpak/flatpak#5073]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.14.0

### What Linux distribution are you using?

Linux Mint

### Linux distribution version

20.3 Mate (based on Ubuntu 20.04)

### What architecture are you using?

x86_64

### How to reproduce

**1st example: Ungoogled Chromium is available today (2022-09-02) in version 105.0.5195.52 at flathub, while 104.0.5112.102 is installed in my system. I type:  "flatpak update com.github.Eloston.UngoogledChromium".
Here is the output:**
"Looking for updates…

        ID                                                      Branch             Op            Remote             Download
 1.     com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub              < 1,1 MB
 2.     com.github.Eloston.UngoogledChromium                    stable             u             flathub            < 133,0 MB

Proceed with these changes to the system installation? [Y/n]:"

**When I type "Y", here is the output:**
"ID                                                      Branch             Op            Remote             Download
        ID                                                      Branch             Op            Remote             Download
 1. [—] com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub                1,5 MB / 1,1 MB
        ID                                                      Branch             Op            Remote             Download
        ID                                                      Branch             Op            Remote             Download
 1. [—] com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub                1,5 MB / 1,1 MB
        ID                                                      Branch             Op            Remote             Download
        ID                                                      Branch             Op            Remote             Download
 1. [✗] com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub              1,5 MB / 1,1 MB
        ID                                                      Branch             Op            Remote             Download
        ID                                                      Branch             Op            Remote             Download
 1. [✗] com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub              1,5 MB / 1,1 MB
 2. [/] com.github.Eloston.UngoogledChromium                    stable             u             flathub            121,0 MB / 133,0 MB

Warning: Could not unmount revokefs-fuse filesystem at /var/tmp/flatpak-cache-MFVBS1/com.github.Eloston.UngoogledChromium.Codecs-T7TSR1: Le processus fils s’est terminé avec le code 1
        ID                                                      Branch             Op            Remote             Download
 1. [✗] com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub              1,5 MB / 1,1 MB
 2. [✗] com.github.Eloston.UngoogledChromium                    stable             u             flathub            121,0 MB / 133,0 MB

Warning: Could not unmount revokefs-fuse filesystem at /var/tmp/flatpak-cache-MFVBS1/com.github.Eloston.UngoogledChromium.Codecs-T7TSR1: Le processus fils s’est terminé avec le code 1
Warning: Could not unmount revokefs-fuse filesystem at /var/tmp/flatpak-cache-MFVBS1/com.github.Eloston.UngoogledChromium.Codecs-T7TSR1: Le processus fils s’est terminé avec le code 1
(internal error, please report)
Warning: Could not unmount revokefs-fuse filesystem at /var/tmp/flatpak-cache-MFVBS1/com.github.Eloston.UngoogledChromium-2C28R1: Le processus fils s’est terminé avec le code 1
Warning: Could not unmount revokefs-fuse filesystem at /var/tmp/flatpak-cache-MFVBS1/com.github.Eloston.UngoogledChromium-2C28R1: Le processus fils s’est terminé avec le code 1
(internal error, please report)
Updates complete."

**Updates failed, old version is still installed:**
"flatpak list
Name                                             Application ID                                        Version                  Branch               Installation
ungoogled-chromium                               com.github.Eloston.UngoogledChromium                  104.0.5112.102           stable               system"

**2nd example: I type:  "flatpak update". Here is the output:**
"Looking for updates…
Required runtime for com.obsproject.Studio/x86_64/stable (runtime/org.kde.Platform/x86_64/6.3) found in remote flathub
Do you want to install it? [Y/n]:"

**When I type Y, here is the output:**
"       ID                                                      Branch             Op            Remote             Download
 1.     com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub              < 1,1 MB
 2.     com.obsproject.Studio.Locale                            stable             u             flathub            < 174,5 kB (partial)
 3.     com.github.Eloston.UngoogledChromium                    stable             u             flathub            < 133,0 MB
 4.     org.kde.Platform.Locale                                 6.3                i             flathub            < 325,8 MB (partial)
 5.     org.kde.Platform                                        6.3                i             flathub            < 247,0 MB
 6.     com.obsproject.Studio                                   stable             u             flathub            < 193,1 MB
 7.     org.kde.okular.Locale                                   stable             u             flathub              < 3,3 MB (partial)
 8.     org.kde.okular                                          stable             u             flathub             < 65,1 MB

Proceed with these changes to the system installation? [Y/n]: "

**When I type "Y", all installations fail with the same error "Could not unmount revokefs-fuse filesystem..." for each installation. And installations are unchanged.**
**The command "flatpak repair" does not detect any problem.**
**Several attempts (after reboot, after manual deleting of "/var/tmp/flatpak-cache-MFVBS1/" don't correct the problem.**

**So, installation update is impossible; installed applications and runtimes do work.**

 

### Expected Behavior

Updates possible

### Actual Behavior

Updates are not possible

### Additional Information

Flatpak 1.14.0 is installed from the flatpak stable PPA.

**Comment by smcv** [comment]
This seems like the same problem as #5071 (a missing dependency on `fuse3`). Workaround: install `fuse3`.

**Comment by MikeNavy** [comment]
****Hi,**

**"fuse3" was installed (during flatpak 1.14.0 installation, fuse3 was required as a dependency by Synaptic and installed).**

**I found the following workaround: I went back to 1.12.7 version. How to:**
- remove flatpak stable PPA from sources
- download the following files "xdg-desktop-portal-gtk_1.8.0-1_flatpak1_20.04_amd64.deb" "xdg-desktop-portal-tests_1.8.1-1_flatpak1_20.04_amd64.deb" "libflatpak0_1.12.7-1flatpak1_20.04_amd64.deb" "gir1.2-flatpak-1.0_1.12.7-1flatpak1_20.04_amd64.deb" "flatpak_1.12.7-1flatpak1_20.04_amd64.deb" from flatpak stable PPA website, see all builds, successful builts.
- then, from downloads directory "sudo dpkg -i *.deb"**

**After that:**
- in Synaptic, "fuse3" appears as unused, in the "can be uninstalled" category,
- the command "flatpak --version" gives the answer "Flatpak 1.12.7",
- **flatpak update is successful:**
"
        ID                                                      Branch             Op            Remote             Download
 1. [✓] com.github.Eloston.UngoogledChromium.Codecs             stable             u             flathub              1,5 MB / 1,1 MB
 2. [✓] com.obsproject.Studio.Locale                            stable             u             flathub              1,5 kB / 174,5 kB
 3. [✓] com.github.Eloston.UngoogledChromium                    stable             u             flathub            126,1 MB / 133,0 MB
 4. [✓] org.kde.Platform.Locale                                 6.3                i             flathub             17,7 kB / 325,8 MB
 5. [✓] org.kde.Platform                                        6.3                i             flathub            129,3 MB / 247,0 MB
 6. [✓] com.obsproject.Studio                                   stable             u             flathub            189,0 MB / 193,1 MB
 7. [✓] org.kde.okular.Locale                                   stable             u             flathub              1,4 kB / 3,3 MB
 8. [✓] org.kde.okular                                          stable             u             flathub             37,3 MB / 65,1 MB

Changes complete."

**Comment by smcv** [comment]
I'm preparing a version of the 1.14.0 backport for focal that uses libfuse version 2 instead of version 3, which will hopefully resolve this.

**Comment by smcv** [comment]
Please try flatpak version `1.14.0-2flatpak1~20.04.1` from the PPA when it becomes available. It's currently in a queue waiting for Launchpad to publish it, but should become available later today.

**Comment by MikeNavy** [comment]
Hi, 

I will test it, but I need the availability of some updates to check updates are successful.

Regards,

MN

**Comment by smcv** [comment]
> I need the availability of some updates to check updates are successful

You could uninstall a package that is not immediately important to you and then reinstall it, or install something small like `org.gtk.Gtk3theme.Greybird` and then uninstall it? As long as you're installing it into the system installation (`flatpak install` without `--user`), that will exercise the same code path that wasn't working for updates.

**Comment by smcv** [comment]
> It's currently in a queue waiting for Launchpad to publish it, but should become available later today

The updated package should be available now.

**Comment by MikeNavy** [comment]
Hi,
**I added flatpak stable PPA to the list of sources, then upgraded.**

**Synaptic installation history:**
`The following packages have been updated:
flatpak (1.12.7-1flatpak1~20.04) to 1.14.0-2flatpak1~20.04.1
gir1.2-flatpak-1.0 (1.12.7-1flatpak1~20.04) to 1.14.0-2flatpak1~20.04.1
libflatpak0 (1.12.7-1flatpak1~20.04) to 1.14.0-2flatpak1~20.04.1`

**The "flatpak --version" command output is now "Flatpak 1.14.0"**

**As suggested, I uninstalled a small package:**
"flatpak uninstall com.virustotal.VirusTotalUploader

        ID                                         Branch          Op
 1.     com.virustotal.VirusTotalUploader          stable          r

Proceed with these changes to the system installation? [Y/n]: 

Uninstall complete"

**Then I re-installed it:**
"flatpak install com.virustotal.VirusTotalUploader
Looking for matches…

com.virustotal.VirusTotalUploader permissions:
    ipc      network      fallback-x11     wayland     x11     dri     file access [1]     dbus access [2]

    [1] host:ro, xdg-config/kdeglobals:ro
    [2] com.canonical.AppMenu.Registrar


        ID                                           Branch           Op           Remote            Download
 1.     com.virustotal.VirusTotalUploader            stable           i            flathub           < 185,7 kB

Proceed with these changes to the system installation? [Y/n]: 

        ID                                           Branch           Op           Remote            Download
 1. [✓] com.virustotal.VirusTotalUploader            stable           i            flathub           265,5 kB / 185,7 kB

Installation complete."

**All was successful. Though this is now an update, if you think it uses the same code, the bug seems to be solved.**
